### ISSUE TITLE
run() command

### ISSUE THREAD

**Issue by h5rdly** [google/python-subprocess32#17]
Hi,

First, thank you for this port. 
I use subprocess32 quite alot, and recently discovered that Python 3.5 actually has a new function named run() which is somewhat more advanced. Would it be possible to add it to subprocess32?

**Comment by gpshead** [comment]
You're not the first person to ask about this.  I'll leave this issue open to track interest in that.

I can't make any good arguments against doing it for API reasons given that I already included 3.3's timeout support in this backport.  So it's mostly a matter of time to pull those parts in, which does not look very complicated:

https://github.com/python/cpython/blob/3.5/Lib/subprocess.py#L319

**Comment by h5rdly** [comment]
Looks complicated to my novice comprehensions, that's why I'm just asking :)

What I like about run() is that it returns an object with everything inside, makes it more comfortable to work with. It would be sweet to be able to use run() and not worry about back compatibility, like you can do with scandir and pathlib2.

**Comment by darkfeline** [comment]
>I'll leave this issue open to track interest in that.

I guess I'll chime in with my support for this.  The old API is terrible in comparison.

**Comment by gpshead** [comment]
Thanks for the PR. I'm looking at it now. Some comments I might've added on the docstring change... but those appear to be minor details I'd like to see cleaned up in the upstream code so I'll take care of those myself in both places after getting this in.

I had resisted back-porting `run()` support initially as it is the second meaningful not 2.7-API backwards compatible change. The first is the timeout support from 3.3 that I added early on. But given it is the second I have no _good_ reason to resist. I've already drifted away from "what was in 3.2" as the unfortunate choice of backport module name implies. :)

My goal is to get this in during pycon over the next week.
